pkg/plugins/manager/pipeline/bootstrap/factory.go

```diff
@@ -23,8 +23,29 @@ func NewDefaultPluginFactory(assetPath *assetpath.Service) *DefaultPluginFactory
 	return &DefaultPluginFactory{assetPath: assetPath}
 }
 
-func (f *DefaultPluginFactory) createPlugin(p plugins.FoundPlugin, class plugins.Class,
+func (f *DefaultPluginFactory) createPlugin(bundle *plugins.FoundBundle, class plugins.Class,
 	sig plugins.Signature) (*plugins.Plugin, error) {
+	plugin, err := f.newPlugin(bundle.Primary, class, sig)
+	if err != nil {
+		return nil, err
+	}
+
+	children := make([]*plugins.Plugin, 0, len(bundle.Children))
```

Comment:
This `children` slice isn't used anywhere else. I think you meant to assign the slice to `plugin.Children` instead ?

```suggestion
	plugin.Children := make([]*plugins.Plugin, 0, len(bundle.Children))
```